### DISCUSSION TITLE
NullPointer region must not be null.

### DISCUSSION THREAD

**Discussion by ntalamdotcom** [aws/aws-sdk-java-v2#5661] [Q&A]
I am unable to understand this problem.
I have used the archetype to work with Java 17, apigatewayv2 as service, I have set the aws credentials as env vars (windows 10).
When running "mvn clean install" I get that error.
Where should I set the region if not by the local credentials??

**Comment by debora-ito** [comment]
Can you show the full stacktrace of the error?
How are you setting the region?

**Comment by ntalamdotcom** [comment]
Hi,
I was not setting the region. 
I was not asked to do it. I thought it was extracting that from my credentials.
I would like to know what is the right way to do it (by default) without writing it in the code

**Reply by debora-ito** [reply]
See [Use AWS Regions](https://docs.aws.amazon.com/sdk-for-java/latest/developer-guide/region-selection.html) in the Developer Guide.
